x-pack/metricbeat/module/aws/s3_request/s3_request.go

```diff
@@ -124,14 +112,20 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) {
 		for _, bucketName := range bucketNames {
 			event, err := createS3RequestEvents(metricDataOutputs, regionName, bucketName)
 			if err != nil {
-				m.logger.Error(err.Error())
+				m.Logger().Error(err.Error())
 				event.Error = err
 				report.Event(event)
 				continue
 			}
-			report.Event(event)
+
+			if reported := report.Event(event); !reported {
+				m.Logger().Debug("Error trying to emit event")
```

Comment:
Nit. These are not errors, the message should be something like "fetch interrupted, metricset stop" or similar.

Reply:
@jsoriano  Good point, I changed it to "Fetch interrupted, failed to emit event". Thank you!